servicetalk-http-api/src/main/java/io/servicetalk/http/api/DelegatingHttpServiceContext.java

```diff
@@ -40,6 +40,16 @@ public DelegatingHttpServiceContext(final HttpServiceContext other) {
         this.delegate = other;
     }
 
+    // public for tests
```

Comment:
can we avoid adding public methods to public APIs just for tests?

Reply:
One of the Netty tests was assuming that the context would be a particular class which instead now gets wrapped by `ExecutionContextOverridingServiceContext`. The easiest resolution was to allow the test to accept either the class it expected or `ExecutionContextOverridingServiceContext` and extract the Netty context from the delegating. Due to module separation the `getDelegate()` method can't be package private.

Reply:
can we adjust the test and avoid adding new APIs?

Reply:
Probably. This was expedient for a draft PR.

Reply:
I didn't find a way to easily fix the `NettyHttpServer` to not require the addition of an accessor to `DelegatingHttpServiceContext`.  It could access the field via reflection if we really did not want to add the `getDelegate()` method.